tests/input-types.test.ts

```diff
@@ -0,0 +1,207 @@
+      INACTIVE
+      PENDING
+    }
+
+    input AddressInput {
+      street: String!
+      city: String!
+      state: String
+      zipCode: String
+      country: String!
+    }
+
+    input UserInput {
+      name: String!
+      email: String!
+      age: Int
+      birthday: Date
+      status: Status
+      isActive: Boolean
+      address: AddressInput
+      tags: [String!]
+      optionalTags: [String]
+      permissions: [String!]!
+      metadata: [String]!
+    }
+
+    input UpdateUserInput {
+      name: String
+      email: String
+      age: Int
+      status: Status
+    }
+
+    input SearchFilters {
+      minAge: Int
+      maxAge: Int
+      statuses: [Status!]
+      isActive: Boolean
+    }
+
+    type User {
+      id: ID!
+      name: String!
+      email: String!
+      age: Int
+      birthday: Date
+      status: Status
+    }
+
+    type Query {
+      # Scalar arguments
+      user(id: ID!): User
+      userByEmail(email: String!): User
+
+      # Optional arguments
+      users(limit: Int, offset: Int): [User!]!
+
+      # Input type arguments
+      searchUsers(filters: SearchFilters): [User!]!
+      getUsersInRange(minAge: Int!, maxAge: Int!): [User!]!
+
+      # Array arguments
+      usersByIds(ids: [ID!]!): [User!]!
+      usersByOptionalIds(ids: [ID]): [User!]!
+    }
+
+    type Mutation {
+      createUser(input: UserInput!): User!
+      updateUser(id: ID!, input: UpdateUserInput!): User!
+
+      # Multiple arguments including inputs
+      createUserWithAddress(
+        userInput: UserInput!
+        addressInput: AddressInput!
+        sendWelcomeEmail: Boolean
+      ): User!
+    }
+  `;
+
+  it("generates input types with proper nullability", async () => {
+    const schema = createTestSchema(schemaWithInputs);
+    const code = await runPlugin(schema, {
+      scalars: { Date: "Date", DateTime: "Date" },
+      mappers: {},
+      enumValues: {},
+    });
+
+    expect(code).toContain(`export interface UserInput {
+  name: string;
+  email: string;
+  age?: number | null | undefined;
+  birthday?: Date | null | undefined;
+  status?: Status | null | undefined;
+  isActive?: boolean | null | undefined;
+  address?: AddressInput | null | undefined;
+  tags?: string[] | null | undefined;
+  optionalTags?: Array<string | null | undefined> | null | undefined;
+  permissions: string[];
+  metadata: Array<string | null | undefined>;
+}`);
+  });
+
+  it("generates nested input types", async () => {
+    const schema = createTestSchema(schemaWithInputs);
+    const code = await runPlugin(schema, {
+      scalars: {},
+      mappers: {},
+      enumValues: {},
+    });
+
+    expect(code).toContain(`export interface AddressInput {
+  street: string;
+  city: string;
+  state?: string | null | undefined;
+  zipCode?: string | null | undefined;
+  country: string;
+}`);
+  });
+
+  it("generates query resolvers with scalar arguments", async () => {
+    const schema = createTestSchema(schemaWithInputs);
```

Comment:
chore: seems like all these `with ... arguments` tests should also be asserting against the generated `QueryFooArgs` interface, not just the resolver signature.